server/jobs/storage_backend.go

```diff
@@ -1,22 +1,95 @@
 package jobs
 
 import (
+	"fmt"
+	"io"
+
+	"github.com/graymeta/stow"
+	"github.com/pkg/errors"
 	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/logging"
 )
 
+const PageSize = 100
+
 //go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_storage_backend.go StorageBackend
 
 type StorageBackend interface {
 	// Read logs from the storage backend. Must close the reader
 	Read(key string) ([]string, error)
 
 	// Write logs to the storage backend
-	Write(key string, logs []string) (success bool, err error)
+	Write(key string, reader io.Reader) (success bool, err error)
+}
+
+type storageBackend struct {
+	location      stow.Location
+	logger        logging.SimpleLogging
+	containerName string
+}
+
+func (s *storageBackend) Read(key string) ([]string, error) {
+	return []string{}, nil
+}
+
+func (s *storageBackend) Write(key string, reader io.Reader) (success bool, err error) {
+	containerFound := false
+
+	// Function to write to container
+	writeFn := func(container stow.Container, err error) error {
+		if err != nil {
+			s.logger.Warn(errors.Wrapf(err, "walking container %s at location: %s", s.containerName, s.location).Error())
```

Comment:
Why not just return error instead of logging?

Reply:
Done!